crates/matrix-sdk/src/event_handler/mod.rs

```diff
@@ -142,15 +151,20 @@ impl HandlerKind {
 }
 
 /// A statically-known event kind/type that can be retrieved from an event sync.
-pub trait SyncEvent {
-    #[doc(hidden)]
+#[doc(hidden)]
+pub trait SyncEvent: Sized {
     const KIND: HandlerKind;
     #[doc(hidden)]
     const TYPE: Option<&'static str>;
```

Comment:
The difference between `KIND` and `TYPE` isn't _super_ clear. They are almost synonym. Can you take the opportunity of this PR to add some comments here please?

Also, why removing `#[doc(hidden)]` on `KIND`, but keep it on `TYPE`?

Reply:
The `#[doc(hidden)]` thing wat was an accident, I'll revert that.